### ISSUE TITLE
[1.12.2] Oil Broken Textures

### ISSUE THREAD

**Issue by vrork** [BuildCraft/BuildCraft#4165]
BuildCraft version: 7.99.17
Forge version: 14.23.3.2690
Link to crash report or log: https://www.dropbox.com/s/fcx0dwso37uy5ib/logs.rar?dl=0 (i cant upload to gist for some reason. Size probably)
Singleplayer or multiplayer: SP
Steps to reproduce: Unknown
Additional information: BC Oil seems to be trying to get a texture that is non-existent from another mod (i think)
Interesting find in debug.log:
https://pastebin.com/aCmuqkiF

**Comment by catwet12** [comment]
From looking over your log your mod list is very similar to my own and i have found the offending mod, aetherworks. Seeing as how this mod was originally released 5 days ago and the issue was brought up 4 days ago, you installed it and then the textures broke, Like you i thought it was another mods oil. yet i had BC, immersive petroleum, and pneumaticraft installed and no texture problems. Yet this morning i added 7 mods; the betweenlands, embers, erebus, magical psi, twilight forest, ember root zoo, and aetherworks. As soon as i found this problem i removed all those mods and added one back at a time and each time checked my broken world, as i added more and more the problem was gone until i installed aetherworks once more. So aetherworks is the offending mod not BC or Immersive petroleum.

**Comment by vrork** [comment]
Ah thanks! I will point this to the Aetherworks mod guy.

**Comment by vrork** [comment]
Linking the finds of the other mod dev:
https://github.com/V0idWa1k3r/Aetherworks/issues/3#issuecomment-390297640

There seems to be some bad interaction beteween buildcraft and his mod but not exclusive to his mod

**Comment by AEnterprise** [comment]
this isn't a mod specific issue, this is because you have 2 mods that add oil, with one being added at a later time, this has caused the default oil fluid of the world and the game instance to desync, causing the world to use an oil that forge did not stitch the textures for

see https://github.com/MinecraftForge/MinecraftForge/issues/4929

**Comment by vrork** [comment]
I am not sure i understand what you mean. 
First: I didnt added any mod that adds oil after i added buildcraft and generated the world. 
Second: The mod linked before (that seems to be causing some kind of problem) doesnt add any oil of any kind.

**Comment by DaedalusGame** [comment]
Yeah this case is odd because it a) happens with WHAT installed (which will stitch all fluid textures) and b) buildcraft stitches the fluid textures always

**Comment by AEnterprise** [comment]
i think there might be 2 different bugs here:
a) while buildcraft always stitches it's textures not all mods do, if another mods oil is the dominant one in the world but you add BC after and that becomes dominant for the game instance the wold will still use the other one that doesn't have it's textures stitched

b) it seems some other mods like WHAT mess with fluid textures in such a way that they make BC's fluid texture stitching mess up

**Comment by DaedalusGame** [comment]
This happens with all Buildcraft Fluids, not just the ones that overlap with other mods, and WHAT will always run after Buildcraft unless event priority is silently discarded for stitch events

**Comment by AlexIIL** [comment]
> as such it might be corrupting (read badly replacing) the textures, causing them to break

It just calls `TextureMap.registerSprite`, which shouldn't do anything if the sprite already exists. That's literally all it does, so I don't think that should be an issue, as registerSprite doesn't do anything if the existing sprite already exists.
https://github.com/DaedalusGame/WHAT/blob/1d5d6b111ed10ce344e8ed276a9343e515b19a01/src/main/java/what/What.java#L13-L29

If anything might be wrong it's buildcraft's code for regsitering sprites, as we do some non-standard things to register the sprite (like load other textures via a custom sprite class).

I think the only useful thing to do would be to actually test this with a debugger, or add a lot of debug log lines when registering to see what's going on.

**Comment by vrork** [comment]
@AEnterprise In my case, i have WHAT disabled, and when i disabled Aetherworks, the textures were all fine

**Comment by vrork** [comment]
Any news on this problem?

**Comment by vrork** [comment]
Guys? please? Any developement on this problem?

**Comment by AlexIIL** [comment]
@vrork it's very unlikely that I'll look at this properly anytime soon.

**Comment by vrork** [comment]
:( buuuh! Why?

**Comment by CvoidWyvern** [comment]
I'm still getting this error on the latest versions of both mods, forge 14.23.5.2836

**Comment by AlexIIL** [comment]
@CvoidWyvern I haven't looked at this yet, and probably won't any soon unfortunately. If or when I do I'll make sure to update this issue.
